### PR TITLE
Fixed CodeQL Azure Pipeline to run correctly

### PR DESCRIPTION
**Pull request by wynonagcc**
- Fixed the indentation
- Changed the commit and ref to check the current commit in the current branch or PR
- Added CodeQL config file to ignore checking the codeql-runner files
- Removed `npm install` as we only need to do `init` and `analyze` in CodeQL for JavaScript

Misc:
- Added additional check for priceRange in listing to prevent type confusion through parameter tampering. After the fix, it still alerts on the same line, so I think it's just a false positive and closed the alert.

### CODE REVIEW

**Comment by devin-efendy** [comment]
Is this only running for `fujiji-server`?

**Comment by wynonagcc** [comment]
> Is this only running for `fujiji-server`?

No, it checks everything other than the one defined in `paths-ignore`

File: codeql-azure-pipeline.yml [hunk 1]
```diff
@@ -17,18 +17,18 @@ steps:
 
 - script: |
     npm install
-    npm test
-  displayName: 'npm install and test'
+  workingDirectory: fujiji-server/
```
Comment (devin-efendy) [inline codeql-azure-pipeline.yml:20]: Ah okay, I thought that was the case because of this lol. Lgtm, then!

**Review by devin-efendy — APPROVED** [review]

File: codeql-azure-pipeline.yml (hunk 1 above)
Comment (wynonagcc) [reply codeql-azure-pipeline.yml:20]: Makes sense, I wonder whether I'm supposed to do npm install for the fujiji-client or not, or even npm install is needed, looking into this!

Comment (wynonagcc) [reply codeql-azure-pipeline.yml]: Turns out we don't even need to do that, just need to do init and analyze in CodeQL for JavaScript

**Pull request merged**